lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
       success: (data) -> cb?(data)
       error: std_ajax_err => @$request_response_error.text gettext "Error changing user's permissions."
 
+class EmailSelectors
+  constructor: (@$container, @$section, params={}) ->
+    params = _.defaults params,
+      label : $container.data 'label'
+
+    template_html = $("#email-list-widget-template").html()
+    @$container.html Mustache.render template_html, params
+    @cur_column = 0
+    @$table = $("#queryTableBody")
+    @labelArray = ($container.data 'selections').split '<>'
+    @$list_selector = @$container.find 'select.single-email-selector'
+    # populate selectors
+    @$list_selector.empty()
+    @$list_endpoint = $container.data 'list-endpoint'
+    @$rolename = $container.data 'rolename'
+    @$list_selector.append $ '<option/>'
+    if this.$container.attr('data-label') == 'Select Section'
+      @load_list()
+    else if this.$container.attr('data-label') == 'Select Problem'
+      @load_list()
+    else
+      for label in @labelArray
+        @$list_selector.append $ '<option/>',
+          text: label
+
+    @$list_selector.change =>
+      $opt = @$list_selector.children('option:selected')
+      return unless $opt.length > 0
```

Comment:
Warning: CoffeeScript rant ahead:

I really don't care for inline/postfix boolean statements. Scanning
through the code, this line appears, at first glance to be an
unconditional return statement. Only after reading nearly halfway
through the line does it become apparent that some cases will fall
through this statement. To exacerbate, since CoffeeScript doesn't
require parens in either boolean statements or function calls, 
this takes an additional context parsing to be sure that it isn't equivalent to
`return unless($opt.length > 0)`.

From a maintainabilty perspective (the whole "code is read more times
than it's written"), I'd prefer to see this as:
```javascript
if (!$opts.length) {
    return;
}
```